networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java

```diff
@@ -237,10 +242,19 @@ public SafeFuture<Void> requestBlockAndBlobsSidecarByRoot(
                     method,
                     new BeaconBlockAndBlobsSidecarByRootRequestMessage(blockRoots),
                     listener))
-        .orElse(
-            SafeFuture.failedFuture(
-                new UnsupportedOperationException(
-                    "BlockAndBlobsSidecarByRoot method is not available")));
+        .orElse(failWithUnsupportedMethodException("BlockAndBlobsSidecarByRoot"));
+  }
+
+  @Override
+  public SafeFuture<Void> requestBlobSidecarsByRoot(
+      final List<BlobIdentifier> blobIdentifiers, final RpcResponseListener<BlobSidecar> listener) {
+    return rpcMethods
+        .blobSidecarsByRoot()
+        .map(
+            method ->
+                requestStream(
+                    method, new BlobSidecarsByRootRequestMessage(blobIdentifiers), listener))
```

Comment:
Where will be validation for us not to send something bigger than allowed in the fork and get ban? 
Also we need to test this

Reply:
The problem is we don't know the current size since we don't have the recent chain data in this class. I wanted to keep it simpler and not include this check, since we only use this method in the RecentBatchFetcher and never make big requests. Wdyt @tbenr . I remember we discussed that.

Reply:
As I understand, we are going to deploy similar mechanism in `byRange` and historical sync for it could be big

Reply:
I see batch sizes are `SYNC_BLOB_SIDECARS_SIZE` but I think we should verify it's under bounds explicitly

Reply:
Yes true, in by range though, we can get the spec from the earliest slot requested and compute the max size. By root, we can't unless we have the RecentChainData

Reply:
humm.. we could add a unit test that checks `SYNC_BLOB_SIDECARS_SIZE` against all forks?

Reply:
Added unit test